### PR TITLE
Allow accumulators to take an AbstractDict

### PR DESCRIPTION
**Pull request by nickrobinson251**

### CODE REVIEW

File: src/accumulator.jl [hunk 1]
```diff
@@ -7,9 +7,10 @@ end
 ## constructors
 
 Accumulator{T, V}() where {T,V<:Number} = Accumulator{T,V}(Dict{T,V}())
+Accumulator(map::AbstractDict) = Accumulator(Dict(map))
 
 counter(T::Type) = Accumulator{T,Int}()
-counter(dct::Dict{T,V}) where {T,V<:Integer} = Accumulator{T,V}(copy(dct))
+counter(dct::AbstractDict{T,V}) where {T,V<:Integer} = Accumulator{T,V}(Dict(dct))
```
Comment (nicoleepp) [inline src/accumulator.jl]:
```suggestion
counter(dct::AbstractDict{T, V}) where {T, V<:Integer} = Accumulator{T, V}(Dict(dct))
```

File: test/test_accumulator.jl [hunk 2]
```diff
@@ -90,6 +90,16 @@
         @test ctm["b"] == 0
     end
 
+    @testset "From AbstractDict" begin
+        kv = (:a => 1, :b => 2)
+        for f in (Accumulator, counter)
+            ct_odict = f(OrderedDict(kv))
+            ct_dict = f(Dict(kv))
+            @test ct_odict isa Accumulator{Symbol,Int}
```
Comment (nicoleepp) [inline test/test_accumulator.jl]:
```suggestion
            @test ct_odict isa Accumulator{Symbol, Int}
```

**Comment by oxinabox** [comment]
LGTM.
please bump versiom number so i can tag release

File: src/accumulator.jl (hunk 1 above)
Comment (oxinabox) [reply src/accumulator.jl]: See also #539

Comment (nickrobinson251) [reply src/accumulator.jl]: i think no space is the dominant style in the file here (except that one case above on L9).

Comment (oxinabox) [reply src/accumulator.jl]: Lets make the change it gets close to the goal of #539

Comment (nickrobinson251) [reply src/accumulator.jl]: let's leave style changes to a different PR (happy to see an issue for it!)

**Pull request merged**